### PR TITLE
Introduce Promisify and use it in the module;

### PR DESCRIPTION
**Pull request by bnjbvr**
(And add a .gitignore)

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -2,15 +2,30 @@ var fs = require('fs');
...
+  return function() {
+    var _this = this;
+    var args = Array.prototype.slice.call(arguments, 0);
+    // Note: "this" is extracted from this scope.
+    return new Promise(function (accept, reject) {
+      // Add the callback function to the list of args
+      args.push(function (err) {
+        var rest = Array.prototype.slice.call(arguments, 1);
+        console.log('in promisified func\n', err, rest);
+        if (typeof err !== 'undefined' && err !== null) {
```
Comment (espadrine) [inline main.js]: This is equivalent to the shorter `if (err != null)`.

Comment (bnjbvr) [inline main.js]: Good catch!

File: main.js [hunk 2]
```diff
@@ -2,15 +2,30 @@ var fs = require('fs');
...
+function promisify(func) {
+  return function() {
+    var _this = this;
+    var args = Array.prototype.slice.call(arguments, 0);
+    // Note: "this" is extracted from this scope.
+    return new Promise(function (accept, reject) {
+      // Add the callback function to the list of args
+      args.push(function (err) {
+        var rest = Array.prototype.slice.call(arguments, 1);
+        console.log('in promisified func\n', err, rest);
```
Comment (espadrine) [inline main.js]: I think it would be preferable to avoid having a log here.

**Comment by espadrine** [comment]
Review done. Thanks for contributing!

**Comment by bnjbvr** [comment]
Thanks for the review! I've addressed comments.

**Comment by espadrine** [comment]
GitHub is drunk. Merged as bd6ac0471eff4a9eac3351458f396d47e3d04843!

**Pull request closed without merging**